src/applications/bmqstoragetool/m_bmqstoragetool_compositesequencenumber.cpp

```diff
@@ -71,24 +71,27 @@ CompositeSequenceNumber::fromString(bsl::ostream&      errorDescription,
     try {
         size_t posFirst, posSecond;
 
-        unsigned long uLong = bsl::stoul(firstPart, &posFirst);
-        d_seqNumber         = bsl::stoul(secondPart, &posSecond);
+        const unsigned long       uLong     = bsl::stoul(firstPart, &posFirst);
+        const bsls::Types::Uint64 seqNumber = bsl::stoul(secondPart,
+                                                         &posSecond);
 
         if (posFirst != firstPart.size() || posSecond != secondPart.size()) {
             throw bsl::invalid_argument("");  // THROW
         }
 
-        d_leaseId = static_cast<unsigned int>(uLong);
-        if (uLong != d_leaseId) {
+        unsigned int leaseId = static_cast<unsigned int>(uLong);
+        if (uLong != leaseId) {
             throw bsl::out_of_range("");  // THROW
         }
 
-        if (d_leaseId == 0 || d_seqNumber == 0) {
+        if (leaseId == 0 || seqNumber == 0) {
             errorDescription << "Invalid input: zero values encountered.";
             return *this;  // RETURN
         }
 
-        d_isSet = true;
+        d_compositeSequenceNumber = bsl::make_pair(leaseId, seqNumber);
+
+        *success = true;
```

Comment:
I think it's possible to rewrite the code by using `do-while(false)` to make it more clear.

Smth like this:
```
do {
    // ...
    if (someError()) {
        errorDescription << "Bad thing happened!";
        break;
    }
    // ...

    *success = true;
    return *this;
} while (false);

*success = false;
return *this;
```

This way:
- No need to "touch" `success` variable twice (it's being reset in the beginning of the function)
- No need to create disposable exception objects